src/background/user-storage.ts

```diff
@@ -2,13 +2,18 @@ import {DEFAULT_SETTINGS, DEFAULT_THEME} from '../defaults';
 import {debounce} from '../utils/debounce';
 import {isURLMatched} from '../utils/url';
 import {UserSettings} from '../definitions';
-import {readSyncStorage, readLocalStorage, writeSyncStorage, writeLocalStorage} from './utils/extension-api';
+import {readSyncStorage, readLocalStorage, writeSyncStorage, writeLocalStorage, isWriting} from './utils/extension-api';
 
 const SAVE_TIMEOUT = 1000;
 
+interface UserStorageOptions {
+    onSettingsChange: () => any;
+}
+
 export default class UserStorage {
-    constructor() {
+    constructor({onSettingsChange}: UserStorageOptions) {
```

Comment:
Maybe let's rename to `onRemoteSettingsChange` or something like this? So it will be clear that the callback is triggered when settings are changed from the outside.